### PR TITLE
refactor: UserSubmenu to JSX and "add button support" :)

### LINKED ISSUE #1510: Add support for button type of User navigation sub item
The [UserSubMenu component](https://github.com/JoinColony/colonyCDapp/blob/master/src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx) currently only has support for external and internal links, button clicks are "hacked" links.
We should add support for different types of components there, maybe use composition instead.

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Rewrite the `UserSubmenu` component to be less config driven and more JSX.
I've reused styles via the tailwind module approach, since all the different links and buttons technically have the same style.
I've also pulled the submenu id into an enum while I was at it.

**Changes** 🏗

* `UserSubmenu` is now a handler that renders one of three menus and uses an enum

Resolves  #1510

### CODE REVIEW

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/blocks/MenuList/MenuList.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { PropsWithChildren } from 'react';
```
Comment (jakubcolony) [inline src/components/common/Extensions/UserNavigation/partials/UserSubmenu/blocks/MenuList/MenuList.tsx:1]: What's the idea with blocks? How are they different from partials?

**Review by jakubcolony — COMMENTED** [review]
I've got a question about the new folder structure. If we end up introducing new patterns, it'd be good to have them in living standards too.

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/blocks/MenuList/MenuList.tsx (hunk 1 above)
Comment (bassgeta) [reply src/components/common/Extensions/UserNavigation/partials/UserSubmenu/blocks/MenuList/MenuList.tsx:1]: My thinking was that 'blocks' are components which are reusable among the `partials`, but come to think of it, we can just put them into the partials directory.

**Comment by jakubcolony** [comment]
@bassgeta might need a rebase before I should reapprove

**Review by willm30 — CHANGES_REQUESTED** [review]
"Base Currency" should be the currency symbol, e.g. "GBP"
Captura de pantalla 2024-01-12 140302

This should be a two-column layout.
Captura de pantalla 2024-01-12 140307

I recommend looking at the relevant files as they are currently on master for reference. Also see the [original pr](https://github.com/JoinColony/colonyCDapp/pull/1572).

**Review by willm30 — APPROVED** [review]
Currency menu looking fine 👍

**Review by jakubcolony — APPROVED** [review]

**Pull request merged**